Don't use reflect in webidl dictionary setters (#3898)
rustwasm/wasm-bindgen: CHANGELOG.md

* Stabilize Web Share API.
  [#3882](https://github.com/rustwasm/wasm-bindgen/pull/3882)

* Don't use reflect in WebIDL dictionary setters. Also, importing getters/setters from JS now supports names containing hyphens by optionally surrounding the name with quotes, e.g. `#[wasm_bindgen(method, setter = "x-cdm-codecs")]`.

>   [#3898](https://github.com/rustwasm/wasm-bindgen/pull/3898)

Reviewer: I think the point here is not supporting hyphens, but strings in general.
It would also be good to mention that Web API bindings are going to be larger now.

Author: Looking at the generated code for `#[wasm_bindgen(method, setter = "x-cdm-codecs")]` I'm realising I really don't understand how it works.. I was never able to find the generated JS code that calls the setter or assigns the value even when closely inspecting a dummy project that makes use of it. But it somehow magically works :shrug: 

Anyways, I updated the changelog as requested

> * Stop using reflect in WebIDL dictionary setters, instead create a shim to allow the wasm_bindgen macro to implement the function in javascript which avoids passing the attribute name string over the WASM/JS boundary. This increases the size of the Web API bindings but should be more efficient. Also, importing getters/setters from JS now supports specifying the JS attribute name as a string, e.g. `#[wasm_bindgen(method, setter = "x-cdm-codecs")]`.

Reviewer: ```suggestion
* Generate JS bindings for WebIDL dictionary setters instead of using `Reflect`. This increases the size of the Web API bindings but should be more performant. Also, importing getters/setters from JS now supports specifying the JS attribute name as a string, e.g. `#[wasm_bindgen(method, setter = "x-cdm-codecs")]`.
```
